### ISSUE TITLE
Atkey namespace overridden by AtClientPreference namespace

### ISSUE THREAD

**Issue by sachins-geekyants** [atsign-foundation/at_client_sdk#841]
### Describe the bug

This happened when we tried deleting wavi image key from Buzz app.

In Buzz, we created following atKey with `wavi` as namespace
```
 atKey = AtKey()
      ..key = "image"
      ..sharedWith = null
      ..namespace = 'wavi'
      ..metadata = metaData;
```

when we called delete operation on it, the namespace was overridden by the one added in `AtClientPreference`.

Thus the actual key was not deleted.

### Steps to reproduce

1. Run Buzz app and try to delete image with above-mentioned code.

### Expected behavior

- the namespace added in atKey should not have been overridden.

### Smartphones

- iPhone 12 simulator

### Were you using an atApplication when the bug was found?

Buzz

**Comment by sitaram-kalluri** [comment]
Found the root cause of the issue and fixed it. Following PR contains the fix: https://github.com/atsign-foundation/at_client_sdk/pull/846

**Comment by sitaram-kalluri** [comment]
The changes are published in at_client v3.0.49
